### PR TITLE
HTCONDOR-2455 remove classad unit suffix

### PR DESCRIPTION
**Pull request by GregThain**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/classad/lexer.cpp [hunk 1]
```diff
@@ -456,20 +456,6 @@ tokenizeNumber (void)
...
-		case 'B': f = Value::B_FACTOR; wind( ); break;	
-		case 'K': f = Value::K_FACTOR; wind( ); break;
-		case 'M': f = Value::M_FACTOR; wind( ); break;
-		case 'G': f = Value::G_FACTOR; wind( ); break;
-		case 'T': f = Value::T_FACTOR; wind( ); break;
-		default:
-			f = Value::NO_FACTOR;
-		}
-	}
-
```
Comment (JaimeFrey) [inline src/classad/lexer.cpp:472]: If we're going to leave variable NumberFactor f here, we should initialize it to Value::NO_FACTOR.

**Review by JaimeFrey — CHANGES_REQUESTED** [review]

File: src/classad/lexer.cpp (hunk 1 above)
Comment (GregThain) [reply src/classad/lexer.cpp:472]: FWIW, my thought is to leave the `NumberFactor` thoughout the code for a release or two, just to make sure we don't need to undo this change in a hurry, then rip out all remnants of this feature.

Comment (JaimeFrey) [reply src/classad/lexer.cpp:472]: I agree with that idea. My main point is that we shouldn't pass around an initialized variable.

**Review by JaimeFrey — APPROVED** [review]
LGTM

**Pull request merged**